### PR TITLE
Add latency test with gatling.

### PR DESCRIPTION
**Pull request by cbickel**
This PR adds a performance test, which measures the latency of warm actions.

The test can be executed for NodeJs, Python, swift and Java actions.

To test the warm latency the test creates an action, invokes it ones warm and invokes it 100 times warm afterwards. In the end of the test the action is deleted again.
Only one action after each other is activated. There are never two actions at the same time in the system.

In addition this PR adds the abilitly to create and invoke actions with the gatling extension for OpenWhisk.

## Related issue and scope
- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components
- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [x] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes
- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

File: performance/README.md [hunk 1]
```diff
@@ -61,3 +61,33 @@ You can run the simulation with (in OPENWHISK_HOME)
 ```
 OPENWHISK_HOST="openwhisk.mydomain.com" CONNECTIONS="10" REQUESTS_PER_SEC="50" ./gradlew gatlingRun-ApiV1Simulation
 ```
+
+##### LatencySimulation
```
Comment (rabbah) [inline performance/README.md]: Latency Simulation? (Two words?)
Please backtick the kinds below.

File: performance/README.md [hunk 2]
```diff
@@ -61,3 +61,33 @@ You can run the simulation with (in OPENWHISK_HOME)
...
+
+This simulation creates actions of the following four kinds: nodejs:default, swift:default, java:default and
+python:default.
+Afterwards the action is invoked once. This is the cold-start and will not be part of the thresholds.
+Next, the action will be invoked 100 times one after each other. The last step is, that the action will be deleted.
+
+Once one language is finished, the next kind will be taken. They are not running in parallel. There are never more than
+1 activations in the system, as we only want to meassure latency of warm activations.
```
Comment (rabbah) [inline performance/README.md]: Is this guaranteed with a blocking invoke then?

Comment (rabbah) [reply performance/README.md]: I think you’re expecting this invariant to hold irrespective of number of controllers or invokers in system. If so can you add this stipulation as well?

File: performance/gatling_tests/src/gatling/resources/data/nodeJSAction.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+function main(params) {
+    var greeting = "Hello" + (params.text || "stranger") + "!";
+    console.log(greeting);
+    return { payload: greeting };
+}
```
Comment (rabbah) [inline performance/gatling_tests/src/gatling/resources/data/nodeJSAction.js]: It seems unnecessary to duplicate these tests aren’t there a consistent set of basic actions for all runtimes in the tests/dat/actions folder?

Comment (cbickel) [reply performance/gatling_tests/src/gatling/resources/data/nodeJSAction.js]: I checked in these actions here to ensure, that all actions do exactly the same. By ensuring this, we can also compare the runtimes against each others.

In the `tests/dat/actions` are a lot of actions, but I didn't find one action, that is doing exactly the same in all languages.

Comment (rabbah) [reply performance/gatling_tests/src/gatling/resources/data/nodeJSAction.js]: Thanks for checking.

**Pull request merged**